extensions/eclipse-che-theia-plugin-remote/src/node/plugin-remote-init.ts

```diff
@@ -362,9 +380,47 @@ class PluginDeployerHandlerImpl implements PluginDeployerHandler {
         }
 
     }
+    async getPluginDependencies(pluginToBeInstalled: PluginDeployerEntry): Promise<PluginDependencies | undefined> {
+        const pluginPath = pluginToBeInstalled.path();
+        try {
+            const manifest = await this.reader.readPackage(pluginPath);
+            if (!manifest) {
+                return undefined;
+            }
+            const metadata = this.reader.readMetadata(manifest);
+            const dependencies: PluginDependencies = { metadata };
+            dependencies.mapping = this.reader.readDependencies(manifest);
+            return dependencies;
+        } catch (e) {
+            console.error(`Failed to load plugin dependencies from '${pluginPath}' path`, e);
+            return undefined;
+        }
+    }
 
-    getPluginMetadata(pluginToBeInstalled: PluginDeployerEntry): Promise<PluginMetadata> {
-        return this.reader.getPluginMetadata(pluginToBeInstalled.path());
+    /**
+     * @throws never! in order to isolate plugin deployment
+     */
+    protected async deployPlugin(entry: PluginDeployerEntry, entryPoint: keyof PluginEntryPoint): Promise<void> {
+        const pluginPath = entry.path();
+        try {
+            const manifest = await this.reader.readPackage(pluginPath);
+            if (!manifest) {
+                return;
+            }
+
+            const metadata = this.reader.readMetadata(manifest);
+            if (this.deployedBackendPlugins.has(metadata.model.id)) {
```

Comment:
Maybe log warning

Reply:
it's also from upstream: 
https://github.com/eclipse-theia/theia/blob/6347c5ccbf016b7cfa70976f69616eb1bb49c5b0/packages/plugin-ext/src/hosted/node/hosted-plugin-deployer-handler.ts#L116